consensus/src/lib.rs

```diff
@@ -50,11 +66,113 @@ pub enum ConsensusCreationError {
     MiningStopped,
     #[error("Mining failed")]
     MiningFailed,
+    #[error("Staking error")]
+    StakingError(#[from] ConsensusPoSError),
+    #[error("Staking failed")]
+    StakingFailed,
+    #[error("Staking stopped")]
+    StakingStopped,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+pub enum GenerateBlockInputData {
```

Comment:
Please add Encode/Decode indexes for enum arms

Reply:
Done.

I wonder if in the future we could add something to `do_checks` that detects codec enums and warns when they don't have an index?